From a34cc85e6516736a4892c42c93fced60056011d0 Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Tue, 28 May 2019 14:54:45 -0400
Subject: [PATCH] Use backports.lzma instead of pyliblzma in setup.py

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index db885ec9..b4fadda7 100755
--- a/setup.py
+++ b/setup.py
@@ -93,7 +93,7 @@ requires = {
         # there is not way to define an alternative here (AFAIK, yoh), we will
         # use pyliblzma as the default for now.  Patch were you would prefer
         # backports.lzma instead
-        'pyliblzma; python_version < "3.3"',
+        'backports.lzma; python_version < "3.3"',
         'simplejson',
         'whoosh',
     ],
-- 
2.21.0.593.g511ec345e18

